pkg/api/v1alpha1/vspheremachineconfig_types.go

```diff
@@ -123,6 +123,13 @@ func (c *VSphereMachineConfig) Validate() error {
 	return validateVSphereMachineConfig(c)
 }
 
+// ValidateHasTemplate verifies that a VSphereMachineConfig object has a template.
+// Specifying a template is required when submitting an object via webhook,
+// as we only support auto-importing templates when creating a cluster via CLI.
+func (c *VSphereMachineConfig) ValidateHasTemplate() error {
+	return validateVSphereMachineConfigHasTemplate(c)
```

Comment:
I really wanted to add this to `VSphereMachineConfig.Validate`, but that is not possible because we use that validate method in the CLI to semantically validate cluster config in the CLI flow.

Writing the following more for myself than anyone else:
My intuition is that we are close on this, but for future direction we may want to make an effort to clearly delineate:
- CLI cluster config semantic validation
- CLI defaulter logic
- Webhook cluster config semantic validation
- Webhook defaulter logic

A lot of things will be shared across those four bullet points.